pynamodb/attributes.py

```diff
@@ -501,33 +508,108 @@ def deserialize(self, value):
 
 class UTCDateTimeAttribute(Attribute):
     """
-    An attribute for storing a UTC Datetime
+    A UTC DateTime attribute
+
+    :param ttl: If True, this attribute is used as the TTL for purposes of automatic item expiry and deletion
+
+        Only a single attribute in a table may be used as the TTL. The first attribute with TTL enabled
+        will be used; repeated use of the flag will be ignored.
+
+        Note that expired items may persist up to two days after their TTL expires. See: 
+        https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateTimeToLive.html
+    """
+    attr_type = NUMBER
+    epoch = datetime.utcfromtimestamp(0).replace(tzinfo=tzutc())
+
+    def __init__(self, hash_key=False, range_key=False, null=None, default=None, attr_name=None, ttl=False):
+        super(UTCDateTimeAttribute, self).__init__(hash_key=hash_key,
+                                                   range_key=range_key,
+                                                   null=null,
+                                                   default=default,
+                                                   attr_name=attr_name)
+        self.is_ttl = ttl
```

Comment:
I'm assuming the answer is yes, but can Dynamo handle non-integer numbers in the TTL attribute?

Reply:
That behavior is not specifically addressed anywhere, but in my tests it simply ignores anything with a non-Number TTL. The following note comes from the feature announcement:

https://aws.amazon.com/blogs/aws/new-manage-dynamodb-items-using-time-to-live-ttl/
> The TTL attribute can be indexed or projected, but it cannot be an element of a JSON document. As I indicated earlier, it must have the Number data type. You can use IAM to regulate access to this attribute, just as you can do for any other one. Items that do not have the designated TTL attribute will not be considered for deletion. In order to avoid a possible accidental deletion due to a malformed TTL value, items that appear to be older than 5 years will not be deleted.

It appears that the implementation is via a background scan process. Records are deleted if the TTL field is:
* Present
* Numeric
* Not within a JSON document
* In the past
* Less than 5 years in the past

Anything else is ignored.

Reply:
Have you tested floating point "Numeric" attributes? (things like `ttl_attribute: {"N": "12345.678"}`)

Reply:
Yes - if you stick a datetime object with millisecond precision into that field, the resulting number is floating-point instead of integer. DynamoDB doesn't expire things with anything approaching millisecond granularity (it's somewhat 'best effort') but it will otherwise handle them correctly.